app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/AbstractSampleProvider.java

```diff
@@ -90,6 +99,29 @@ public void addGBActivitySamples(T[] activitySamples) {
         getSampleDao().insertOrReplaceInTx(activitySamples);
     }
 
+    @Override
+    public void exportToCSV(T[] activitySamples, File outFile) {
+        String separator = ",";
+
+        LOG.info("Exporting samples into csv file: " + outFile.getName());
+        try {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(outFile));
```

Comment:
Please use
`try (BufferedWriter bw = ....) {`
and remove the `bw.close()`
That way the writer will always be closed (also when an exception occurs).

Reply:
I didn't know this try block syntax. So, it's a try-with-resources statement. It closes bw at end of execution. I understand so why you remove bw.close().